test/rubygems/test_gem_specification.rb

```diff
@@ -3090,6 +3090,28 @@ def test_validate_license
     WARNING
   end
 
+  def test_validate_nil_license
+    util_setup_validate
+
+    use_ui @ui do
+      @a1.license = nil
+      @a1.validate
+    end
+
+    assert_empty @ui.error
+  end
+
+  def test_validate_licenses_containing_multiple_types
+    util_setup_validate
+
+    e = assert_raise Gem::InvalidSpecificationException do
+      @a1.licenses = [nil, "MIT"]
+      @a1.validate
+    end
+
+    assert_match "licenses must be of the same class, detected 2 classes", e.message
```

Comment:
This message is somewhat confusing to me. Could we say something more clear and specific. Something like "Licenses field cannot include nil with other licenses."

I haven't thought about the phrasing very much but I'd like it to be more specific about nil not being allowed in the licenses array unless it is the only element.

Reply:
Based on conversations below, removed this check and associated test